cloudinary.go

```diff
@@ -0,0 +1,391 @@
+	}
+	return nil
+}
+
+// Path returns the full path for a file.
+func (r *Cloudinary) Path(file string) string {
+	return validPath(file)
+}
+
+// Put stores a new file on the disk.
+func (r *Cloudinary) Put(file, content string) error {
+	tempFile, err := r.tempFile(content)
+	defer os.Remove(tempFile.Name())
+	if err != nil {
+		return err
+	}
+	_, err = r.instance.Upload.Upload(r.ctx, tempFile.Name(), uploader.UploadParams{
+		PublicID:       r.getPublicId(file),
+		UseFilename:    api.Bool(true),
+		UniqueFilename: api.Bool(false),
+		ResourceType:   r.getResourceType(file),
+	})
+	return err
+}
+
+// PutFile stores a new file on the disk.
+func (r *Cloudinary) PutFile(path string, source filesystem.File) (string, error) {
+	return r.PutFileAs(path, source, source.File())
+}
+
+// PutFileAs stores a new file on the disk.
+func (r *Cloudinary) PutFileAs(path string, source filesystem.File, name string) (string, error) {
+	uploadResult, err := r.instance.Upload.Upload(r.ctx, source.File(), uploader.UploadParams{
+		Folder:         validPath(path),
+		PublicID:       r.getPublicId(name),
+		UseFilename:    api.Bool(true),
+		UniqueFilename: api.Bool(false),
+	})
+	if err != nil {
+		return "", err
+	}
+	return uploadResult.PublicID, nil
+}
+
+// Size returns the file size of a given file.
+func (r *Cloudinary) Size(file string) (int64, error) {
+	resource, err := r.getAsset(file)
+	if err != nil {
+		return 0, err
+	}
+	return int64(resource.Bytes), nil
+}
+
+// TemporaryUrl get the temporary url of a file.
+func (r *Cloudinary) TemporaryUrl(file string, time time.Time) (string, error) {
+	return "", errors.New("cloudinary doesn't support temporary url")
+}
+
+// WithContext sets the context for the driver.
+func (r *Cloudinary) WithContext(ctx context.Context) filesystem.Driver {
+	driver, err := NewCloudinary(ctx, r.config, r.disk)
+	if err != nil {
+		color.Redf("[Cloudinary] init disk error: %+v\n", err)
+		return nil
+	}
+	return driver
+}
+
+// Url returns the url for a file.
+func (r *Cloudinary) Url(file string) string {
+	asset, err := r.getAsset(file)
+	if err != nil {
+		return ""
+	}
+	return asset.SecureURL
+}
+
+func (r *Cloudinary) getAsset(path string) (*uploader.ExplicitResult, error) {
+	return r.instance.Upload.Explicit(r.ctx, uploader.ExplicitParams{
+		PublicID:     r.getPublicId(path),
+		Type:         "upload",
+		ResourceType: r.getResourceType(path),
+	})
+}
+
+func (r *Cloudinary) tempFile(content string) (*os.File, error) {
+	tempFile, err := os.CreateTemp(os.TempDir(), "goravel-")
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := tempFile.WriteString(content); err != nil {
+		return nil, err
+	}
+
+	return tempFile, nil
+}
+
+func (r *Cloudinary) getResourceType(path string) string {
+	extension := strings.TrimPrefix(filepath.Ext(path), ".")
+	value := "image"
+	resourceTypes := r.config.Get(fmt.Sprintf("filesystems.disks.%s.resource_types", r.disk), defaultResourcesTypes()).(map[string][]string)
```

Comment:
There seems to be no change here. Based on the below image or range the ResourceTypes. What's your choice?

![telegram-cloud-photo-size-5-6246531548778182330-y](https://github.com/goravel/cloudinary/assets/24771476/a4a31fba-219d-44a4-b96b-d162edeb95a3)